torch/_prims/nvfuser_prims.py

```diff
@@ -299,13 +304,71 @@ def _amin_nvfuser(
 _nvfuser_impls["transpose"] = _transpose_nvfuser
 _nvfuser_impls["squeeze"] = _squeeze_nvfuser
 _nvfuser_impls["view_of"] = _view_of_nvfuser
+_nvfuser_impls["rand_like"] = _rand_like_nvfuser
 _nvfuser_impls["sum"] = _sum_nvfuser
 _nvfuser_impls["var"] = _var_nvfuser
 _nvfuser_impls["var_mean"] = _var_mean_nvfuser
 _nvfuser_impls["amax"] = _amax_nvfuser
 _nvfuser_impls["amin"] = _amin_nvfuser
 
 
+def register_rand_like():
+    name = "rand_like"
+
+    nvprim.define(
+        "rand_like(Tensor self, *, ScalarType? dtype=None, Layout? layout=None, "
+        + "Device? device=None, bool? pin_memory=None, MemoryFormat? memory_format=None) -> Tensor"
+    )
+
+    def _meta_rand_like(
+        self,
+        *,
+        dtype=None,
+        layout=None,
+        device=None,
+        pin_memory=None,
+        memory_format=None,
+    ):
+        strides = make_contiguous_strides_for(self.shape)
```

Comment:
these are not correct strides for rand_like?

Reply:
We currently don't support maintaining strided outputs in the nvFuser integration.  Given that restriction, is this still incorrect to do?

Reply:
We never get the bandwidth to plumb memory format support through. For now, codegen only produces contiguous tensor. Luckily nvfuser_prims are only used by nvfuser, so we are not breaking others.

There were conversations about respecting memory format and that is on our radar. For now I think we can just leave a TODO here.